src/main/java/net/glowstone/entity/GlowEntity.java

```diff
@@ -605,6 +615,40 @@ public void pulse() {
                 }
             }
         }
+        GlowBlock block = world.getBlockAt(location);
+        if (block.getType().equals(Material.PORTAL)
+            && BlockPortal.getBoundingBox(block).intersects(boundingBox)) {
+            if (!isInPortal) {
+                EventFactory.getInstance().callEvent(
+                    new EntityPortalEnterEvent(this, block.getLocation()));
+                isInPortal = true;
+            }
+            if (server.getAllowNether() && portalCooldown <= 0) {
+                GlowWorld w = getWorld().getEnvironment().equals(Environment.NETHER)
+                    ? server.getWorld("world") : server.getWorld("world_nether");
+                TravelAgent agent = w.getTravelAgent();
+                boolean destIsNether = w.getEnvironment().equals(Environment.NETHER);
+                int destX = destIsNether ? location.getBlockX() / 8 : location.getBlockX() * 8;
+                int destY = destIsNether ? location.getBlockY() / 8 : location.getBlockY() * 8;
+                int destZ = destIsNether ? location.getBlockZ() / 8 : location.getBlockZ() * 8;
+                Location requested = new Location(w, destX, destY, destZ);
+                if (agent.getCanCreatePortal() || agent.findPortal(requested) != null) {
+                    Location teleportLocation = agent.findOrCreate(requested);
+                    EntityPortalEvent p = EventFactory.getInstance().callEvent(
```

Comment:
Avoid 1-letter variable. Rename this to `event` and the GlowWorld instance to `world`.

Reply:
Applied